src/main/java/com/powsybl/openloadflow/network/impl/Networks.java

```diff
@@ -106,36 +106,74 @@ public static List<LfNetwork> load(Network network, LfNetworkParameters paramete
         return LfNetwork.load(network, new LfNetworkLoaderImpl(), parameters);
     }
 
-    public static List<LfNetwork> load(Network network, SlackBusSelector slackBusSelector, Reporter reporter) {
-        return LfNetwork.load(network, new LfNetworkLoaderImpl(), slackBusSelector, reporter);
-    }
-
     public static List<LfNetwork> load(Network network, LfNetworkParameters parameters, Reporter reporter) {
         return LfNetwork.load(network, new LfNetworkLoaderImpl(), parameters, reporter);
     }
 
-    public static List<LfNetwork> load(Network network, LfNetworkParameters networkParameters,
-                                       Set<Switch> switchesToOpen, Set<Switch> switchesToClose, Reporter reporter) {
+    static class LfNetworkListImpl implements LfNetworkList {
+
+        private final List<LfNetwork> list;
+
+        LfNetworkListImpl(List<LfNetwork> list) {
+            this.list = Objects.requireNonNull(list);
+        }
+
+        @Override
+        public List<LfNetwork> getList() {
+            return list;
+        }
+
+        @Override
+        public void close() {
+            // nothing to clean
+        }
+    }
+
+    static class LfNetworkListVariantCleanupImpl implements LfNetworkList {
```

Comment:
Wouldn't it be clearer/simpler to use if we had only one class? with `tmpVariantId == null` if we didn't clone the working variant, to expose clearly why there's nothing to clean.

Reply:
Variant clean requires 3 attributes (network, old and new tmp variant), it would not be clean to only rely on one of them to decide if we must clean or not. I agree that it would be better to only have one LfNetworkList so I tried another design with just LfNetworkList and an optional embedded object VariantCleaner that is responsible for variant cleaning.